ggml.c

```diff
@@ -4303,10 +4303,43 @@ int64_t ggml_nrows(const struct ggml_tensor * tensor) {
 }
 
 size_t ggml_nbytes(const struct ggml_tensor * tensor) {
-    size_t nbytes = tensor->ne[0]*tensor->nb[0]/ggml_blck_size(tensor->type);
+    // original:
+    //size_t nbytes = tensor->ne[0]*tensor->nb[0]/ggml_blck_size(tensor->type);
+    //for (int i = 1; i < GGML_MAX_DIMS; ++i) {
+    //    nbytes += (tensor->ne[i] - 1)*tensor->nb[i];
+    //}
+    //return nbytes;
+
+    // TODO: the imlpementation below is stupid - need something better
+
+    // sort ne and nb
+    int64_t sne[GGML_MAX_DIMS];
+    size_t  snb[GGML_MAX_DIMS];
+
+    for (int i = 0; i < GGML_MAX_DIMS; ++i) {
+        sne[i] = tensor->ne[i];
+        snb[i] = tensor->nb[i];
+    }
+
+    for (int i = 0; i < GGML_MAX_DIMS; ++i) {
+        for (int j = i + 1; j < GGML_MAX_DIMS; ++j) {
+            if ((snb[i] > snb[j]) || (snb[i] == snb[j] && sne[i] < sne[j])) {
+                size_t tmp = snb[i];
+                snb[i] = snb[j];
+                snb[j] = tmp;
+
+                int64_t tmp2 = sne[i];
+                sne[i] = sne[j];
+                sne[j] = tmp2;
+            }
+        }
+    }
+
+    size_t nbytes = (sne[0]/ggml_blck_size(tensor->type))*snb[0];
     for (int i = 1; i < GGML_MAX_DIMS; ++i) {
-        nbytes += (tensor->ne[i] - 1)*tensor->nb[i];
+        nbytes += (sne[i] - 1)*snb[i];
     }
+
```

Comment:
So, the goal of my implementation was to calculate the address of the last element plus one. However, the implementation assumes that `nb[0] == type_size`, so it doesn't work with transposed tensors. This should fix it for `blck_size == 1`:
```C
size_t nbytes = ggml_type_size(tensor->type);
for (int i = 0; i < GGML_MAX_DIMS; ++i) { 
    nbytes += (tensor->ne[i] - 1)*tensor->nb[i]; 
}
```

However, this will not work with quantized types. A possible solution could be fall back to the previous implementation for `blck_size > 1`, but it would be nicer to have a single implementation.
```C
size_t ggml_nbytes(const struct ggml_tensor * tensor) { 
    size_t nbytes;
    size_t blck_size = ggml_blck_size(tensor->type);
    if (blck_size == 1) { 
        nbytes = ggml_type_size(tensor->type);
        for (int i = 0; i < GGML_MAX_DIMS; ++i) { 
            nbytes += (tensor->ne[i] - 1)*tensor->nb[i]; 
        }
    }
    else {
        nbytes = tensor->ne[0]*tensor->nb[0]/blck_size;
        for (int i = 1; i < GGML_MAX_DIMS; ++i) { 
            nbytes += (tensor->ne[i] - 1)*tensor->nb[i]; 
        }
    }
    return nbytes;
}
```